src/main/groovy/io/seqera/wave/util/RegHelper.groovy

```diff
@@ -221,6 +222,14 @@ class RegHelper {
         try {
             final yaml = new Yaml().load(spackFileContent) as Map
             final spack = yaml.spack as Map
+
+            if( !spack ){
+                throw new BadRequestException("Section \"spack:\" not found in the Spack environment file, please check it and try again")
+            }
+            if( !spack.specs ){
+                throw new BadRequestException("Section \"spack:specs:\" not found in the Spack environment file, please check it and try again")
+            }
```

Comment:
This is a kind of useless, because the exception is thrown in the `catch` statement and the method just return null. Therefore the error message is never returned the user making the request